src/target/riscv/riscv.h

```diff
@@ -272,13 +270,13 @@ struct riscv_info {
 	bool *reserved_triggers;
 
 	/* Memory access methods to use, ordered by priority, highest to lowest. */
-	int mem_access_methods[RISCV_NUM_MEM_ACCESS_METHODS];
+	riscv_mem_access_method_t mem_access_methods[RISCV_MEM_ACCESS_MAX_METHODS_NUM];
+
+	unsigned int mem_access_method_num;
```

Comment:
If you like:

This can be renamed to `num_enabled_mem_access_methods` which is longer but hopefully clearer.

The original name could be interpreted by a reader as "number (ID) of a method".

```suggestion
	unsigned int num_enabled_mem_access_methods;
```

Reply:
Renamed

Reply:
Renamed